gobblin-utility/src/main/java/gobblin/util/AvroUtils.java

```diff
@@ -146,12 +148,31 @@
     }
 
     if ((field + 1) == pathList.size()) {
+      if (data instanceof java.util.HashMap) {
+        return getObjectFromMap((HashMap) data, pathList.get(field));
```

Comment:
Cast it to a `Map` instead of `HashMap`. Since the cast is a raw `Map` you may need to add `@SuppressWarnings("rawtypes")` to the method.